### PR TITLE
Use ID or Where to instead directly use Get when load object from database

### LINKED ISSUE #11921: The usage of xorm get has risk
## Description
Such as the usage of xorm get in models/attachment.go, if the id or uuid is empty, the query sql will be no where: select ... from attachment limit 1.  The query will not be an error, but the query result could be not the result you want.
I suppose we use xorm get by this way:
attach := new(Attachment)
has, err := engine.Where("id=?", id).Get(attach)

## Screenshots
image

### PR DESCRIPTION
**Pull request by lunny**
As title. Fix #11921

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

File: models/branches.go [hunk 1]
```diff
@@ -240,8 +240,8 @@ func getProtectedBranchBy(e Engine, repoID int64, branchName string) (*Protected
 
 // GetProtectedBranchByID getting protected branch by ID
 func GetProtectedBranchByID(id int64) (*ProtectedBranch, error) {
-	rel := &ProtectedBranch{ID: id}
-	has, err := x.Get(rel)
+	rel := &ProtectedBranch{}
+	has, err := x.Where("id=?", id).Get(rel)
```
Comment (6543) [inline models/branches.go]:
```suggestion
	has, err := x.ID(id).Get(rel)
```

File: models/branches.go [hunk 2]
```diff
@@ -509,9 +509,9 @@ func (repo *Repository) GetDeletedBranches() ([]*DeletedBranch, error) {
 }
 
 // GetDeletedBranchByID get a deleted branch by its ID
-func (repo *Repository) GetDeletedBranchByID(ID int64) (*DeletedBranch, error) {
-	deletedBranch := &DeletedBranch{ID: ID}
-	has, err := x.Get(deletedBranch)
+func (repo *Repository) GetDeletedBranchByID(id int64) (*DeletedBranch, error) {
+	deletedBranch := &DeletedBranch{}
+	has, err := x.Where("id=?", id).Get(deletedBranch)
```
Comment (6543) [inline models/branches.go]:
```suggestion
	has, err := x.ID(id).Get(deletedBranch)
```

File: models/user_mail.go [hunk 3]
```diff
@@ -71,8 +71,8 @@ func GetEmailAddresses(uid int64) ([]*EmailAddress, error) {
 // GetEmailAddressByID gets a user's email address by ID
 func GetEmailAddressByID(uid, id int64) (*EmailAddress, error) {
```
Comment (6543) [inline models/user_mail.go:72]: if we get it by ID why we ned the uid?

Comment (lafriks) [reply models/user_mail.go:72]: because in same cases id is provided from query and we need to check that this email address belongs to authorized user

Comment (6543) [reply models/user_mail.go:72]: Only think this was **miss used** for permission chek - we schould do this elsewhere

Comment (lunny) [reply models/user_mail.go:72]: To confirm `uid` is the owner of the record with the `id`.

Comment (6543) [reply models/user_mail.go:72]: this will be another refactor ... but out of scope of this pull :)

**Review by 6543 — APPROVED** [review]

**Comment by 6543** [comment]
🚀

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
Please send backports

**Comment by 6543** [comment]
Backports:
 * #11934
 * #11935

**Pull request merged**